src/main/kotlin/org/rust/lang/core/psi/ext/RsLitExpr.kt

```diff
@@ -22,26 +25,53 @@ import org.rust.lang.core.psi.impl.RsExprImpl
 import org.rust.lang.core.psi.kind
 import org.rust.lang.core.stubs.RsLitExprStub
 import org.rust.lang.core.stubs.RsPlaceholderStub
-import org.rust.lang.core.stubs.RsStubLiteralType
+import org.rust.lang.core.stubs.RsStubLiteralKind
 import org.rust.lang.core.types.ty.TyFloat
 import org.rust.lang.core.types.ty.TyInteger
 
-val RsLitExpr.stubType: RsStubLiteralType? get() {
+val RsLitExpr.stubKind: RsStubLiteralKind? get() {
     val stub = (stub as? RsLitExprStub)
-    if (stub != null) return stub.type
+    if (stub != null) return stub.kind
     val kind = kind
     return when (kind) {
-        is RsLiteralKind.Boolean ->  RsStubLiteralType.Boolean
-        is RsLiteralKind.Char -> RsStubLiteralType.Char(kind.isByte)
-        is RsLiteralKind.String -> RsStubLiteralType.String(kind.value?.length?.toLong(), kind.isByte)
-        is RsLiteralKind.Integer -> RsStubLiteralType.Integer(TyInteger.fromSuffixedLiteral(integerLiteral!!))
-        is RsLiteralKind.Float -> RsStubLiteralType.Float(TyFloat.fromSuffixedLiteral(floatLiteral!!))
+        is RsLiteralKind.Boolean ->  RsStubLiteralKind.Boolean(kind.value)
+        is RsLiteralKind.Char -> RsStubLiteralKind.Char(kind.value, kind.isByte)
+        is RsLiteralKind.String -> RsStubLiteralKind.String(kind.value, kind.isByte)
+        is RsLiteralKind.Integer -> RsStubLiteralKind.Integer(kind.value, TyInteger.fromSuffixedLiteral(integerLiteral!!))
+        is RsLiteralKind.Float -> RsStubLiteralKind.Float(kind.value, TyFloat.fromSuffixedLiteral(floatLiteral!!))
         else -> null
     }
 }
 
-val RsLitExpr.integerLiteralValue: String? get() =
-    (stub as? RsLitExprStub)?.integerLiteralValue ?: integerLiteral?.text
+val RsLitExpr.booleanValue: Boolean?
+    get() {
+        val stubKind = stubKind ?: return (kind as? RsLiteralKind.Boolean)?.value
```

Comment:
Looks like if `stubKind` is `null`, then `kind` cannot be `RsLiteralKind.Boolean`. AFAIU, `stubKind` is `null` iff `kind` is `null` or `kind` is not `RsLiteralKind.<...>`.